commit 8d72247daeb36087120046cc39817b7ce85a3703
Author: Albert Astals Cid <aacid@kde.org>
Date:   Thu Apr 18 00:21:31 2019 +0200

    Fix uninitialized memory use when reading malformed tar files
    
    Summary:
    Don't try to read longlink if readLongLink fails
    
    oss-fuzz #14310
    
    Reviewers: apol
    
    Reviewed By: apol
    
    Subscribers: apol, kde-frameworks-devel
    
    Tags: #frameworks
    
    Differential Revision: https://phabricator.kde.org/D20641

diff --git a/src/ktar.cpp b/src/ktar.cpp
index 398e5ac..8890dd9 100644
--- a/src/ktar.cpp
+++ b/src/ktar.cpp
@@ -274,15 +274,16 @@ qint64 KTar::KTarPrivate::readHeader(char *buffer, QString &name, QString &symli
         if (strcmp(buffer, "././@LongLink") == 0) {
             char typeflag = buffer[0x9c];
             QByteArray longlink;
-            readLonglink(buffer, longlink);
-            switch (typeflag) {
-            case 'L':
-                name = QFile::decodeName(longlink.constData());
-                break;
-            case 'K':
-                symlink = QFile::decodeName(longlink.constData());
-                break;
-            }/*end switch*/
+            if (readLonglink(buffer, longlink)) {
+                switch (typeflag) {
+                case 'L':
+                    name = QFile::decodeName(longlink.constData());
+                    break;
+                case 'K':
+                    symlink = QFile::decodeName(longlink.constData());
+                    break;
+                }/*end switch*/
+            }
         } else {
             break;
         }/*end if*/
